### PR TITLE
feat(modal): add support for custom windowClass on modals

### PR DESCRIPTION
**Pull request by colinmorelli**
Adds the ability to add a custom class to a modal when it's opened.

Note that this does not (yet) add support for adding a custom backdrop class. That may need more discussion to talk about what should happen when, for example, multiple modals are open. Whose backdrop gets applied?

Fixes (at least part of) #818

### CODE REVIEW

File: demo/src/app/components/modal/demos/customclass/modal-customclass.ts [hunk 1]
```diff
@@ -0,0 +1,33 @@
+import {Component, ViewEncapsulation} from '@angular/core';
+
+import {NgbModal, ModalDismissReasons} from '@ng-bootstrap/ng-bootstrap';
+
+@Component({
+  selector: 'ngbd-modal-customclass',
+  templateUrl: './modal-customclass.html',
+  encapsulation: ViewEncapsulation.None,
```
Comment (colinmorelli) [inline demo/src/app/components/modal/demos/customclass/modal-customclass.ts]: I don't love disabling view encapsulation on this, but I wanted to keep the associated styles close to the component. Modals are tough in that (by design) they render content outside of the component's view hierarchy.

File: demo/src/app/components/modal/demos/customclass/modal-customclass.ts [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+  styleUrls: [ './modal-customclass.css' ]
+})
+export class NgbdModalCustomClass {
+  closeResult: string;
+
+  constructor(private modalService: NgbModal) {}
+
+  open(content) {
+    this.modalService.open(content, { windowClass: 'dark-modal' }).result.then((result) => {
+      this.closeResult = `Closed with: ${result}`;
```
Comment (pkozlowski-opensource) [inline demo/src/app/components/modal/demos/customclass/modal-customclass.ts]: I don't think we need to repeat all the demo here. Just `this.modalService.open(...)` should be enough, IMO

**Comment by colinmorelli** [comment]
Apologies if I'm missing something obvious but I'm not seeing comments anywhere

**Comment by pkozlowski-opensource** [comment]
> Apologies if I'm missing something obvious but I'm not seeing comments anywhere

@colinmorelli no, it is me failing to push the right buttons :-/

File: src/modal/modal-window.ts [hunk 3]
```diff
@@ -15,7 +15,7 @@ import {ModalDismissReasons} from './modal-dismiss-reasons';
 @Component({
   selector: 'ngb-modal-window',
   host: {
-    'class': 'modal fade in',
+    '[class]': '"modal fade in" + (windowClass ? " " + windowClass : "")',
```
Comment (colinmorelli) [inline src/modal/modal-window.ts]: This is part of the host selectors applied to the angular template. AFAIK there's no way to use ES6 string concatenation here. It follows the same pattern that's used on line 26: `<div [class]="'modal-dialog' + (size ? ' modal-' + size : '')" role="document">`

Comment (pkozlowski-opensource) [inline src/modal/modal-window.ts]: You are right, sorry for the noise.

**Comment by realappie** [comment]
Really a shame this commit only supports a string for the window class property. It would have been more helpful if it was named windowClass and accepted an array of strings !

**Pull request closed without merging**